.github/workflows/convention-develocity-maven-extension-verification.yml

```diff
@@ -0,0 +1,58 @@
+name: Verify Convention Develocity Maven Extension
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build Convention Develocity Maven Extension
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up JDK 8
+        uses: actions/setup-java@v4
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build with Maven
+        working-directory: convention-develocity-maven-extension/extension
+        run: ./mvnw clean install
+      - name: Upload extension
+        uses: actions/upload-artifact@v4
+        with:
+          name: convention-develocity-maven-extension
+          path: ~/.m2/repository/com/myorg
+
+  verification:
+    name: Verify Example Build for Maven ${{ matrix.maven-version }}
+    needs: [build]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        maven-version: ['3.6', '3.8', '3.9']
```

Comment:
I question if we really need samples for these three versions. The samples are exactly the same except for the Maven wrappers. If we really wanted to cross version test these (and we don't do this for the Gradle equivalent), we could just run with different Maven installations. What do you think about either:
* Changing `examples` to `example` and just making the sample project a top-level entry in that folder
* Making a `maven_3` example folder that stays up to date

Reply:
This is a good point. #1 is what you had done originally and in the verification step you updated the wrapper of the 1 example to the different versions. In my efforts to align with the Gradle plugin I made explicit version folders, but I actually really like the idea of just having a `maven_3` folder. I think that actually aligns better with the Gradle plugin. 

That being said, I think we should still test the different versions, but I agree we don't need explicit folders when they are fundamentally the same. Perhaps I can find a good middle ground.

Reply:
Check out https://github.com/gradle/develocity-build-config-samples/pull/636/files/3adfc10d68291c414aca35915dda327a59c5cb3b..00d9cb72abc2872bac795c2d581280eb8850d98e

And: https://github.com/gradle/develocity-build-config-samples/actions/runs/9130461722/job/25107417741?pr=636